### PR TITLE
Include thresholds for ideal heatercooler #617

### PR DESCRIPTION
**Pull request by DaJansenGit**
- integrate into templates
- integrate into usecond
- integrate into json save and load functions

### CODE REVIEW

**Comment by DaJansenGit** [comment]
further todos:
- [ ] make this feature selectable
- [ ] turn off feature for zones without AHU

**Comment by DaJansenGit** [comment]
**Bigger changes:**
I moved the ideal heater and cooler together with the new threshold controller from `AixLib.ThermalZones.ReducedOrder.Multizone.MultizoneEquipped` into `AixLib.ThermalZones.ReducedOrder.ThermalZone.ThermalZoneEquipped` because it's a zone dependent model. It also makes the implementation easier because the threshold controller can easily be switched between existing and not existing during compilation time for every zone separately.  

**Behaviour and way of implementation:**
- default value is threshold usage off
- usage conditions: if with ahu the threshold usage is on, if no ahu threshold usage is off
- if one tries to set threshold usage to true and no ahu exists in this zone, an error is raised

**Comment by MichaMans** [comment]
@DaJansenGit first of all thanks!

The threshold values, is there a sort of reference for that? At the moment, the values look very strict. e.g. 22°C for cooling and 18°C for heating. Are very near the set temperatures. But if we have a reference for that. I'm fine with it.

File: teaser/data/input/teaserjson_input.py [hunk 1]
```diff
@@ -140,7 +140,9 @@ def load_teaser_json(path, project):
             ]
             tz.use_conditions.typical_width = zone_in["use_conditions"]["typical_width"]
             tz.use_conditions.with_heating = zone_in["use_conditions"]["with_heating"]
+            tz.use_conditions.T_threshold_heating = zone_in["use_conditions"]["T_threshold_heating"]
             tz.use_conditions.with_cooling = zone_in["use_conditions"]["with_cooling"]
+            tz.use_conditions.T_threshold_cooling = zone_in["use_conditions"]["T_threshold_cooling"]
             tz.use_conditions.persons = zone_in["use_conditions"]["persons"]
             tz.use_conditions.ratio_conv_rad_persons = zone_in["use_conditions"][
                 "ratio_conv_rad_persons"
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply teaser/data/input/teaserjson_input.py]: you are correct! This was solved

**Pull request merged**